azurerm/resource_arm_api_management_api.go

```diff
@@ -0,0 +1,432 @@
+								string(apimanagement.SwaggerLinkJSON),
+								string(apimanagement.WadlLinkJSON),
+								string(apimanagement.WadlXML),
+								string(apimanagement.Wsdl),
+								string(apimanagement.WsdlLink),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+
+						"wsdl_selector": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"service_name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"endpoint_name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"protocols": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type:             schema.TypeString,
+					DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(apimanagement.ProtocolHTTP),
+						string(apimanagement.ProtocolHTTPS),
+					}, true),
+				},
+				Optional: true,
+				Computed: true, // Azure API sets protocols to https by default
+			},
+
+			"subscription_key_parameter_names": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"header": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"query": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"soap_pass_through": {
+				Type:     schema.TypeBool,
+				Default:  false,
+				Optional: true,
+			},
+
+			"revision": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"version_set_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"is_current": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"is_online": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmApiManagementApiCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).apiManagementApiClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM API Management API creation.")
+
+	resGroup := d.Get("resource_group_name").(string)
+	serviceName := d.Get("service_name").(string)
+	name := d.Get("name").(string)
+
+	//Currently we don't support revisions, so we use 1 as default
+	apiId := fmt.Sprintf("%s;rev=%d", name, 1)
+	d.Set("api_id", apiId)
+
+	var properties *apimanagement.APICreateOrUpdateProperties
+
+	_, hasImport := d.GetOk("import")
+
+	// If import is used, we need to send properties to Azure API in two operations.
+	// First we execute import and then updated the other props.
```

Comment:
Agree with the simplification, not sure about `IsNewResource`... It could be my understanding of how TF manages this, but at least in Azure the resource (API) can change schema without recreating the resource (.e.g. you can import a new Swagger file to an existing API). My understanding is then that this should be able to import multiple times.

Reply:
in which case the `IsNewResource` isn't needed (I was assuming that the schema for a given API versions was immutable/required a version bump)